services/console-proxy/server/src/main/java/com/cloud/consoleproxy/ConsoleProxyNoVncClient.java

```diff
@@ -113,32 +114,36 @@ public void run() {
                             if (client.isVncOverWebSocketConnectionOpen()) {
                                 updateFrontEndActivityTime();
                             }
-                            connectionAlive = client.isVncOverWebSocketConnectionAlive();
-                            try {
-                                Thread.sleep(1);
-                            } catch (Exception e) {
-                                s_logger.warn("Error on sleep for vnc over websocket", e);
-                            }
+                            connectionAlive = session.isOpen();
+                            Thread.sleep(1);
                         } else if (client.isVncOverNioSocket()) {
                             byte[] bytesArr;
                             int nextBytes = client.getNextBytes();
                             bytesArr = new byte[nextBytes];
                             client.readBytes(bytesArr, nextBytes);
+                            s_logger.trace(String.format("Read [%s] bytes from client [%s]", nextBytes, clientId));
```

Comment:
should this be wrapped under isTraceEnabled (or similar)?

Reply:
I think that using `isTraceEnabled` is bad for readability. Furthermore, as we have a Log4J patch prepared. As soon as we merge that patch, none of these checks will be needed anymore.